src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -54,14 +59,17 @@ export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
 
   const languageOptions: Array<{ label: string; value: string }> = [];
 
-  languageService.getLanguages().forEach((language) => {
-    if (
-      (language && props.appName && !language.editorSupportedAppNames?.includes(props.appName)) ||
-      languageService.getUserQueryLanguageBlocklist().includes(language?.id)
-    )
-      return;
-    languageOptions.unshift(mapExternalLanguageToOptions(language!));
-  });
+  languageService
+    .getLanguages()
+    .filter((language) => datasetSupportedLanguages.includes(language.id))
+    .forEach((language) => {
+      if (
+        (language && props.appName && !language.editorSupportedAppNames?.includes(props.appName)) ||
+        languageService.getUserQueryLanguageBlocklist().includes(language?.id)
+      )
+        return;
+      languageOptions.unshift(mapExternalLanguageToOptions(language!));
```

Comment:
Seems you don't need non-null assertion here
```suggestion
      languageOptions.unshift(mapExternalLanguageToOptions(language));
```

Reply:
removed